grails-app/services/com/unifina/service/CommunityJoinRequestService.groovy

```diff
@@ -8,26 +8,20 @@ import com.unifina.domain.community.CommunityJoinRequest
 import com.unifina.domain.community.CommunitySecret
 import com.unifina.domain.security.IntegrationKey
 import com.unifina.domain.security.SecUser
-import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder
-import org.springframework.security.crypto.password.PasswordEncoder
-
-class CommunityProductService {
-	PasswordEncoder encoder = new BCryptPasswordEncoder()
 
+class CommunityJoinRequestService {
 	void onApproveJoinRequest(CommunityJoinRequest c) {
-		c.state = CommunityJoinRequest.State.ACCEPTED
-		// Fetch joinPartStream id from smart contract at address
-		// Backend produces join message to joinPartStream
+		// TODO: Fetch joinPartStream id from smart contract at address
```

Comment:
Can @isentropy assist in creating a method to fetch this information from the contract as well? Very similar to fetching the operator address.

Reply:
ok, I'll add to this branch. @kare feel free to move static helper methods outside Web3jHelper.